collect_app/src/main/java/org/odk/collect/android/formmanagement/FormNavigator.kt

```diff
@@ -18,16 +19,45 @@ class FormNavigator(
     private val instancesRepositoryProvider: () -> InstancesRepository
 ) {
 
-    fun editInstance(activity: Activity, instanceId: Long) {
-        val uri = InstancesContract.getUri(projectId, instanceId)
-        activity.startActivity(
-            Intent(activity, FormEntryActivity::class.java).also {
+    fun editInstance(context: Context, instanceId: Long) {
+        val editingDisabled = !settingsProvider.getProtectedSettings().getBoolean(KEY_EDIT_SAVED)
+        val status = instancesRepositoryProvider().get(instanceId)?.status
+
+        context.startActivity(
+            editInstanceIntent(context, projectId, instanceId, editingDisabled, status)
+        )
+    }
+
+    fun newInstance(context: Context, formId: Long) {
+        context.startActivity(
+            newInstanceIntent(context, projectId, formId)
+        )
+    }
+
+    companion object {
```

Comment:
These static methods would traditionally live on the Activity (`FormEntryActivity` in this case). I opted to keep them here to avoid adding any more code to `FormEntryActivity`.